src/table/components/withSelections.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useContextSelector, TableContext } from '../context';
+import { getSelectionStyle } from '../utils/styling-utils';
+import { getCellSelectionState, SelectionActions, TSelectionActions } from '../utils/selections-utils';
+import { hocProps, SelectionState } from '../../types';
+
+export default function withSelections(CellComponent: (props: hocProps) => JSX.Element) {
+  const HOC = (props: hocProps) => {
+    const { cell, styling, announce } = props;
+    const cellSelectionState = useContextSelector(TableContext, (value) =>
+      getCellSelectionState(cell, value.selectionState as SelectionState)
+    );
+    const selectionDispatch = useContextSelector(
+      TableContext,
+      (value) => value.selectionDispatch as React.Dispatch<TSelectionActions>
+    );
```

Comment:
This isn't super pretty with the casting. The other options is to sanity check them